src/v2i-hub/ODEForwardPlugin/src/CommunicationModeHelper.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <unordered_map>
+#include <string>
+#include <boost/algorithm/string.hpp>
+#include <tmx/TmxException.hpp>
+
+using tmx::TmxException;
+
+namespace ODEForwardPlugin
+{
+    /**
+     * @brief Communication mode: UDP or Kafka
+     */
+    enum class CommunicationMode{
+        UDP,
+        KAFKA
+    };
+
+    class CommunicationModeHelper
+    {
+    private:
+        //Indicator for the communication mode
+        bool _isKAFKAMode = false;
+        bool _isUDPMode = false;
+
+        //Map for communication mode to string
+        std::unordered_map<CommunicationMode, std::string> _communicationModeMap = {
+            {CommunicationMode::UDP, "UDP"},
+            {CommunicationMode::KAFKA, "KAFKA"}
+        };
+    public:
+        CommunicationModeHelper()=default;
+        ~CommunicationModeHelper()=default;
+        /**
+         * @brief Compare the communication mode with the given mode
+         * @param modeSource The communication mode to be compared
+         * @param modeDest The communication mode to be compared with
+         * @return true if the communication mode matches, false otherwise
+         * @throws TmxException if the communication mode is unknown
+         */
+        bool compareCommunicationMode(std::string& modeSource, CommunicationMode modeDest);
+        /**
+         * @brief Set the mode (UDP or KAFKA) of the communication
+         * @param modeSource The communication mode to be set
+         * @throws TmxException if the communication mode is unknown
+         */
+        void setMode(std::string& modeSource);
+        /**
+         * @brief Get the Kafka mode
+         * @return true if the communication mode is Kafka, false otherwise
+         */
+        bool getKafkaMode() const;
```

Comment:
There is only one configuration parameter but the way this is implemented seems to suggest that both code be true simultaneously. Is there a reason why we made 2 boolean variables instead of 1 enumeration?

Reply:
Yeah, enum should work too.

Reply:
https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/656/commits/90a1d5cb362264e3b490242a73f163fda3a695b7